src/rewriters/Rewritings.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2024 Martin Blicha <martin.blicha@gmail.com>
+ *
+ *  SPDX-License-Identifier: MIT
+ *
+ */
+
+#ifndef OPENSMT_REWRITINGS_H
+#define OPENSMT_REWRITINGS_H
+
+#include "ArithLogic.h"
+#include "PTRef.h"
+
+#include <optional>
+
+namespace opensmt::rewritings {
```

Comment:
```suggestion
namespace opensmt {
```
since almost all the functions begin with `rewrite`
It is also better to use e.g. `opensmt::rewriteDistincts` than
```
using namespace opensmt::rewritings;
rewriteDistincts(...);
```

Reply:
See my comment about single namespace vs. organizing the codebase into sub-namespaces.

Reply:
I personally prefer to use sub-namespaces a lot, even to such an extent that the meaning of the function/class depends on the outer namespaces. For example, I could do
```
namespace opensmt::rewrite {
    PTRef distincts(Logic & logic, PTRef fla);
    PTRef distinctsKeepTopLevel(Logic & logic, PTRef fla);
    ...
}
```
which avoids redundancy in the code.
However, here in OpenSMT it is not used like this and such a style would currently be inconsistent with the rest.

Hence I rather recommend to use
```
namespace opensmt {
    PTRef rewriteDistincts(Logic & logic, PTRef fla);
    PTRef rewriteDistinctsKeepTopLevel(Logic & logic, PTRef fla);
    ...
}
```
which avoids redundant code at least a bit.